test-cloud/test-cloud-remote/src/main/java/org/kie/cloud/integrationtests/runners/SmokeTestRunner.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.cloud.integrationtests.runners;
+
+import org.junit.Ignore;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.internal.runners.model.EachTestNotifier;
+import org.junit.runner.Description;
+import org.junit.runner.notification.RunNotifier;
+import org.junit.runner.notification.StoppedByUserException;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+import org.junit.runners.model.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SmokeTestRunner extends BlockJUnit4ClassRunner {
+
+    private static final Logger logger = LoggerFactory.getLogger(SmokeTestRunner.class);
+
+    private static final int RETRY_COUNT = 3;
+    private int failedAttempts = 0;
+
+    public SmokeTestRunner(Class<?> klass) throws InitializationError {
+        super(klass);
+    }
+
+    @Override
+    public void run(final RunNotifier notifier) {
+        EachTestNotifier testNotifier = new EachTestNotifier(notifier, getDescription());
+        Statement statement = classBlock(notifier);
+        try {
+            statement.evaluate();
+        } catch (AssumptionViolatedException e) {
+            testNotifier.fireTestIgnored();
+        } catch (StoppedByUserException e) {
+            throw e;
+        } catch (Throwable e) {
+            retry(testNotifier, statement, e);
+        }
+    }
+
+    @Override
+    protected void runChild(final FrameworkMethod method, RunNotifier notifier) {
+        Description description = describeChild(method);
+        if (method.getAnnotation(Ignore.class) != null) {
+            notifier.fireTestIgnored(description);
+        } else {
+            runTestUnit(methodBlock(method), description, notifier);
+        }
+    }
+
+    protected final void runTestUnit(Statement statement, Description description, RunNotifier notifier) {
+        EachTestNotifier eachNotifier = new EachTestNotifier(notifier, description);
+        eachNotifier.fireTestStarted();
+        try {
+            statement.evaluate();
+        } catch (AssumptionViolatedException e) {
+            eachNotifier.addFailedAssumption(e);
+        } catch (Throwable e) {
```

Comment:
I should not retry on any kind of exception, but only when one exception was thrown. We could create a custom exception for this.
Also, if we finally use the AwaitilityUtils, we could use the .conditionEvaluationListener method to raise this custom exception only when required. 
What do you think?